src/mutation.cpp

```diff
@@ -100,20 +100,13 @@ std::string enum_to_string<mutagen_technique>( mutagen_technique data )
 
 bool Character::has_trait( const trait_id &b ) const
 {
-    mutation_filter filter = mutation_filter::all;
-    if( b.is_valid() ) {
-        if( b.obj().debug ) {
-            filter = mutation_filter::debug;
-        } else if( !b.obj().anger_relations.empty() ) {
-            filter = mutation_filter::anger_relations;
-        } else if( !b.obj().ignored_by.empty() ) {
-            filter = mutation_filter::ignored_by;
-        } else if( !b.obj().social_mods.empty() ) {
-            filter = mutation_filter::social_mods;
-        }
-    }
-    for( const trait_id &mut : get_mutations( true, filter ) ) {
-        if( mut == b ) {
+    for( const std::pair<const trait_id, trait_data> &t : my_mutations ) {
+        if( t.first == b ) {
+            return true;
+        }
+    }
+    for( const trait_id &ench_trait : enchantment_cache->get_mutations() ) {
+        if( ench_trait == b ) {
             return true;
         }
     }
```

Comment:
```suggestion
    return enchantment_cache->get_mutations().count( b );
```

Also, currently `enchantment_cache->get_mutations()` is returning a copy; it should probably return by const reference instead.